lib/delayed/backend/ironmq.rb

```diff
@@ -77,16 +77,18 @@ def save!
         end
 
         def destroy
-          if @msg
+          if @msg && @msg.reserved_count > @timeout_retries
+             @msg = ironmq.queue(queue_name).get_message(@msg.id)
```

Comment:
If condition is true move a message into error queue, you mean mark a message and move in the next pick?